### PR TITLE
Select Device tree based on system

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**
The commit implement changes to select appropriate device tree and JSON based on underneath system.

The JSON is selected based in the IM and HW value that is fetched from the VPD and JSON is having entry to the corresponding device tree for that system.

### CODE REVIEW

File: include/worker.hpp [hunk 1]
```diff
@@ -26,8 +30,14 @@ class Worker
 
     /**
      * @brief Constructor.
+     *
+     * Constructor will also, based on symlink pick the correct JSON and
+     * initialize the parsed JSON variable.
+     *
+     * Note: Throws std::exception in case of construction failure. Caller needs
+     * to handle to detect successfull object creation.
```
Comment (GiridhariKrishna) [inline include/worker.hpp:38]: nit

File: include/worker.hpp [hunk 2]
```diff
@@ -40,7 +50,8 @@ class Worker
      * This API based on system chooses corresponding device tree and JSON.
      * If device tree change is required, it updates the "fitconfig" and reboots
      * the system. Else it is NOOP.
-     * Similarly for JSON, synlink is craeted if not found else ignored.
+     *
+     * Note: In case of any error, exceptionis thrown. Caller need to handle.
```
Comment (GiridhariKrishna) [inline include/worker.hpp:54]: nit

**Review by GiridhariKrishna — COMMENTED** [review]
Yet to complete, just for comments uptill now

File: include/worker.hpp (hunk 1 above)
Comment (SunnySrivastava1984) [reply include/worker.hpp:38]: What is the comment here?

Comment (GiridhariKrishna) [reply include/worker.hpp:38]: Not significant, just the spelling of successful. just for info.

File: include/worker.hpp (hunk 2 above)
Comment (SunnySrivastava1984) [reply include/worker.hpp]: Done.

File: include/worker.hpp (hunk 1 above)
Comment (SunnySrivastava1984) [reply include/worker.hpp]: Done.

File: src/worker.cpp [hunk 3]
```diff
@@ -1,13 +1,266 @@
...
+        if (pos + 1 < entry.size())
+        {
+            fitConfigValue = entry.substr(pos + 1);
+        }
+    }
+
+    return fitConfigValue;
+}
+
+bool Worker::isMotherboardPathOnDBus() const
```
Comment (GiridhariKrishnan) [inline src/worker.cpp:73]: Will this function be used anywhere else, apart from setDevicetreeandJson()? If only there then may be a check should be good, I think in that function.

File: src/worker.cpp [hunk 4]
```diff
@@ -1,13 +1,266 @@
...
+{
+    const std::string& mboardPath =
+        m_parsedJson["frus"][SYSTEM_VPD_FILE_PATH].at(0).value("inventoryPath",
```
Comment (GiridhariKrishnan) [inline src/worker.cpp:76]: Not sure, if it'll throw exception if inventoryPath not there/or should we consider that?

File: src/worker.cpp (hunk 3 above)
Comment (SunnySrivastava1984) [reply src/worker.cpp:73]: What kind of check?

File: src/worker.cpp (hunk 4 above)
Comment (SunnySrivastava1984) [reply src/worker.cpp:76]: No it will not throw any exception.

File: src/worker.cpp (hunk 3 above)
Comment (GiridhariKrishna) [reply src/worker.cpp:73]: motherboard path check

Comment (SunnySrivastava1984) [reply src/worker.cpp:73]: That is what we are doing in this function. You want me to put all these code in that function itself?

Comment (GiridhariKrishnan) [reply src/worker.cpp:73]: ok

File: src/worker.cpp (hunk 4 above)
Comment (GiridhariKrishnan) [reply src/worker.cpp:76]: Ok

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**